extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts

```diff
@@ -17,14 +17,38 @@ export class CheApiServiceImpl implements CheApiService {
 
     private workspaceRestAPI: IRemoteAPI | undefined;
 
+    async getCurrentWorkspaceId(): Promise<string> {
+        return this.getWorkspaceIdFromEnv();
+    }
+
+    async getUserPreferences(filter?: string): Promise<Preferences> {
+        const apiClient = await this.getCheApiClient();
+        return await apiClient.getUserPreferences(filter);
+    }
+
+    async updateUserPreferences(update: Preferences): Promise<Preferences> {
+        const apiClient = await this.getCheApiClient();
+        return await apiClient.updateUserPreferences(update);
+    }
+
+    async replaceUserPreferences(preferences: Preferences): Promise<Preferences> {
+        const apiClient = await this.getCheApiClient();
+        return await apiClient.replaceUserPreferences(preferences);
+    }
+
+    async deleteUserPreferences(list?: string[]): Promise<void> {
+        const apiClient = await this.getCheApiClient();
+        return await apiClient.deleteUserPreferences(list);
+    }
+
     async currentWorkspace(): Promise<cheApi.workspace.Workspace> {
         try {
             const workspaceId = process.env.CHE_WORKSPACE_ID;
             if (!workspaceId) {
                 return Promise.reject('Cannot find Che workspace id, environment variable "CHE_WORKSPACE_ID" is not set');
             }
 
-            const wsClient = await this.wsClient();
+            const wsClient = await this.getCheApiClient();
```

Comment:
probably rename wsClient as well in that case ?

Reply:
done